Make the search by typing feature in ModList more intelligent

slikts opened the pull request:
This makes `ModList` behave more like typical data lists and makes it significantly easier to use because it's possible to go directly to a mod by typing in the first few letters of its name. The delay before it forgets the previous key presses is 1 second. The previous implementation wasn't very useful in cases where a number of mod names share the same first letter.

slikts committed:
Make the search by typing feature in the mod list more intelligent:
search for different keypress combinations and cycle through matches if pressing the same key

RichardLake commented on Main.cs:
```diff
@@ -523,14 +529,44 @@ private void ModList_KeyPress(object sender, KeyPressEventArgs e)
+            if (interval.TotalSeconds < 1 && key != this.lastSearchKey) {
+                // Last keypress was < 1 sec ago and it was a different key, so combine the last and current keys
+                key = this.lastSearchKey + key;
+            }
```
Why wouldn't you allow sequential key presses? There are mods which have "ff" in their names for example.

slikts replied:
Fixed it now; just added a check for there being only one distinct key pressed

slikts committed:
Fixing repeated keypress behavior; cycles through a single key if only
one distinct key is pressed, but searches for the full combination if
more than one key is pressed

slikts committed:
Fix newlines

slikts committed:
Only toggle the checkbox if the current row is visually selected; before it would check it even if the selection was cleared
